tests/integration/test_linspace.py

```diff
@@ -15,46 +15,277 @@
 
 import numpy as np
 import pytest
+from utils.generators import broadcasts_to, mk_seq_array
 
 import cunumeric as num
 
 
-def test_basic():
-    x = np.linspace(2.0, 3.0, num=5)
-    y = num.linspace(2.0, 3.0, num=5)
-    assert np.array_equal(x, y)
+def equivalent_shapes_gen(shape):
+    yield shape
+    for i in range(len(shape) - 1):
+        if shape[i] == 1:
+            i += 1
+            yield shape[i:]
+        else:
+            break
+
+
+@pytest.mark.parametrize(
+    "retstep", (True, False), ids=lambda retstep: f"(retstep={retstep})"
+)
+@pytest.mark.parametrize(
+    "endpoint", (True, False), ids=lambda endpoint: f"(endpoint={endpoint})"
+)
+@pytest.mark.parametrize(
+    "number", (0, 1, 10), ids=lambda number: f"(num={number})"
+)
+@pytest.mark.parametrize(
+    "values",
+    ((10, -5.5), (2.0, 3.0), (0, 0), (1 + 2.5j, 10 + 5j), (0j, 10)),
+    ids=lambda values: f"(values={values})",
+)
+def test_scalar_basic(values, number, endpoint, retstep):
+    start, stop = values
+    x = np.linspace(
+        start, stop, num=number, endpoint=endpoint, retstep=retstep
+    )
+    y = num.linspace(
+        start, stop, num=number, endpoint=endpoint, retstep=retstep
+    )
+
+    if retstep:
+        assert np.array_equal(x[0], y[0])
+        if np.isnan(x[1]) and np.isnan(y[1]):
+            pass
+        else:
+            assert x[1] == y[1]
+    else:
+        assert np.array_equal(x, y)
```

Comment:
`retstep` changes the return type fundamentally. I would suggest avoiding conditionals like this in tests and splitting  with/without retsep cases in to separate tests (here and below)